FirebaseMLModelDownloader/Sources/ModelInfoRetriever.swift

```diff
@@ -281,45 +281,107 @@ class ModelInfoRetriever {
                                     message: ModelInfoRetriever.DebugDescription
                                       .modelInfoUnmodified,
                                     messageCode: .modelInfoUnmodified)
+              self.telemetryLogger?.logModelInfoRetrievalEvent(eventName: .modelDownload,
+                                                               status: .modelInfoRetrievalSucceeded,
+                                                               errorCode: .noError)
               completion(.success(.notModified))
             case 400:
-              let description = ModelInfoRetriever.ErrorDescription.invalidModelName(self.modelName)
-              DeviceLogger.logEvent(level: .debug,
-                                    message: description,
-                                    messageCode: .invalidModelName)
+              if let data = data,
+                let responseJSON = try? JSONSerialization
+                .jsonObject(with: data, options: []) as? [String: Any],
+                let error = responseJSON["error"] as? [String: Any],
+                let errorMessage = error["message"] as? String {
+                DeviceLogger.logEvent(level: .debug,
+                                      message: errorMessage,
```

Comment:
Another thought - would it be easier and more universal to print the response body as a string without any  content parsing.

Reply:
Hm that would be ideal. But I tried using `String(data:, encoding:)` and could an ugly string with a whole bunch of `\n` characters. Is there a better way?

Reply:
Ah, I see. Maybe it's because Swift specifics. This approach worked for be pretty well from Objective-C.

I guess you may print entire `responseJSON` object as a compromise.

Reply:
You could always use that string and then use `.replacingOccurrences(of: "\n", with: " ")` and see if that works :)

Reply:
Gotcha. I ended up using option 1 above - method to reduce duplication, because I realized we don't want the entire error message to be surfaced, just the "message" in the error message. :)